src/ray/gcs/redis_context.h

```diff
@@ -91,19 +91,19 @@ class RedisCallbackManager {
         : callback_(callback),
           is_subscription_(is_subscription),
           start_time_(start_time),
-          io_service_(io_service) {}
+          io_service_(&io_service) {}
 
     void Dispatch(std::shared_ptr<CallbackReply> &reply) {
       std::shared_ptr<CallbackItem> self = shared_from_this();
       if (callback_ != nullptr) {
-        io_service_.post([self, reply]() { self->callback_(std::move(reply)); });
+        io_service_->post([self, reply]() { self->callback_(std::move(reply)); });
       }
     }
 
     RedisCallback callback_;
     bool is_subscription_;
     int64_t start_time_;
-    boost::asio::io_service &io_service_;
+    boost::asio::io_service *io_service_;
```

Comment:
Actually could we use std::shared_ptr<> here instead? That should have minimal overhead (paid once at construction time), and avoids a bunch of hazards with lifetime management.

Reply:
Sure, if you'd like to push the change! I'm not familiar with the code here; just trying to get compile errors out of the way at the moment.